### PR TITLE
F#111 user login server

### PR DESCRIPTION
**Pull request by iylim**
works on #111

### CODE REVIEW

**Comment by glenflorendo** [comment]
Branched from f#94-tmdbapi
Dependent on PR #1

File: config/passport.js [hunk 1]
```diff
@@ -1,38 +1,33 @@
 const passport = require('passport');
 const GoogleStrategy = require('passport-google-oauth20').Strategy;
-// const User = require('../models/user');
+const User = require('../models/user');
```
Comment (glenflorendo) [inline config/passport.js:3]:
```js
const User = require('../models')
```

File: models/index.js [hunk 2]
```diff
@@ -1,16 +1,31 @@
...
+mongoose.connection.on('connected', () =>
+  console.log(`Connected to ${process.env.DB_NAME}`),
+);
+
+mongoose.connection.on('disconnected', () =>
+  console.log(`Disconnected from ${process.env.DB_NAME}`),
+);
 
 module.exports.Movie = require('./movie');
 module.exports.Genre = require('./genre');
```
Comment (glenflorendo) [inline models/index.js:31]: Add
```
module.exports.User = require('./user');
```

File: config/passport.js [hunk 3]
```diff
@@ -1,38 +1,33 @@
...
   clientID: process.env.GOOGLE_CLIENT_ID,
   clientSecret: process.env.GOOGLE_SECRET,
   callbackURL: process.env.GOOGLE_CALLBACK,
-}, function(accessToken, refreshToken, profile, cb) {
-  User.findOne({ googleId: profile.id }, function(err, user) {
-    if (err) return cb(err);
+}, (accessToken, refreshToken, profile, done) => {
+  User.findOne({ googleId: profile.id }, (err, user) => {
+    if (err) return done(err);
     if (user) {
```
Comment (glenflorendo) [inline config/passport.js]: One line for consistency with line 11:
```js
if (user) return done(null, user);
```

File: controllers/userController.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+const jwt = require('jsonwebtoken');
+const User = require('../models/user');
+
+const { SECRET } = process.env;
+
+function createJWT(user) {
+  return jwt.sign({ user }, SECRET, { expiresIn: '24h' });
```
Comment (glenflorendo) [inline controllers/userController.js:7]: Implicit return
```js
createJWT = user => jwt.sign({ user }, SECRET, { expiresIn: '24h' }};
```

File: controllers/userController.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+// module.exports.findOne = (req, res) => {
+//   User.find({ googleId: req.params.id }).then(userData => res.json(userData));
+// };
+
+/**
+ * creates new user in database
+ * @param {Object} req user data
+ * @param {string} res token
+ */
+function signup(req, res) {
```
Comment (glenflorendo) [inline controllers/userController.js:19]: Arrow function

File: controllers/userController.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    })
+    .catch(err => res.status(400).json(err));
+}
+
+/**
+ * Checks user credentials
+ * @param {Object} req credentials
+ * @param {string} res token
+ */
+function login(req, res) {
```
Comment (glenflorendo) [inline controllers/userController.js:36]: Arrow function

**Review by glenflorendo — CHANGES_REQUESTED** [review]
Minor changes. Also, remember to pull before you push.

**Review by glenflorendo — APPROVED** [review]

**Comment by samsaksfithian** [comment]
Works on yamovie/client#111

**Pull request merged**